src/ethereum/frontier/rlp.py

```diff
@@ -485,6 +495,49 @@ def encode_log(raw_log_data: Log) -> Bytes:
     )
 
 
+def decode_to_header(encoded_header: Bytes) -> Header:
+    """
+    Decodes a rlp encoded byte stream assuming that the decoded data
+    should be of type `Header`.
+
+    NOTE - This function is valid only till the London Hardfork. Post that
+    there would be changes in the Header object as well as this function with
+    the introduction of `base_fee` parameter.
```

Comment:
```suggestion
```

I think we can omit this note, since we're going to duplicate the whole directory for every hardfork anyway.